tools/localnet/design.md

```diff
@@ -0,0 +1,95 @@
+responsibility of dev team grows every day. That's why we need good tools helping us to build our confidence that new
+feature works well before pull request is even created by the author.
+
+Before IBC era, testing new features locally was pretty easy. The only required component was a single `sifnode`.
+Now, after releasing IBC it is much more complicated, and many interconnected components are needed.
+For ease of use it seems reasonable to create a tool starting and managing all of them.
+
+## Requirements
+
+To be able to play with all th features of `sifnode` (including IBC) we need these components:
+- at least two `sifnodes` running two separate chains
+- relayer between them
+- local eth testnet
+- peggy between eth and sifchain
+- block explorer for every chain
+
+User selects which blockchains are started.
+
+User chooses if block explorers are started.
+
+Every time we add new supported blockchain `localnet` is extended by adding testnet, relayer and block explorer.
+
+### Sifchain
+
+To support sifchains, 4 validators and 1 full node per chain are started.
+
+It would be nice to implement some logs filter in `sifnode` to prevent regular `tendermint` consensus-related logs
+from being printed in `localnet` unless they are errors or panics. By doing this, relevant logs printed by our app are
+more meaningful and take less space on screen.
+
+[`hermes`](https://hermes.informal.systems/) is configured to relay tokens between two sifchains.
+
+[`Big Dipper`](https://github.com/forbole/big-dipper) is used for exploring blocks.
+
+`sifnode` and `hermes` are single-file pieces of software. Both may be run locally easily, so they don't require
+containers. On the other hand `Big Dipper` requires frontend and `mongo` instance, that's why it's better to execute them
+in containers (ready to use images are available there).
+
+### Ethereum
+
+Single [`geth`](https://github.com/ethereum/go-ethereum) instance is used to create ethereum testnet. It's a single executable so may be used directly,
+without container. 
+
+`peggy` is started to maintain token transmission between ethereum and sifchain. Have to discuss it with peggy
+team.
+
+[`BlockScout`](https://github.com/blockscout/blockscout) block explorer is started for ethereum. Again, it's better to containerize it.
+
+## Tools
+
+### tmate
+
+[`tmate`](https://tmate.io/) is a great tool based on [`tmux`](https://github.com/tmux/tmux/) allowing us to run many
+applications in a single terminal window. Keyboard shortcuts are used to navigate between started apps so all the logs
+may be inspected manually. `tmate` is an extension over `tmux`, supporting remote access to sessions created by others.
+
+So instead of running many terminals, one per component, all of them may be started in a single one making our life easier.
+At the same time, `tmate` provides the list of all running apps, marking those which exited (due to panic for ex.).
+
+### podman
```

Comment:
Is podman Linux only?

Reply:
There's no podman for Windows, there are some tricks to run it on macos. Ok, so we will suport both docker and podman. It's easy.